### PR TITLE
dev12.0 afc 254

### PR DESCRIPTION
**Pull request by robch2342**

### CODE REVIEW

File: auth_saml_ol/controllers/main.py [hunk 1]
```diff
@@ -168,14 +167,17 @@ def signin(self, **kw):
         redirect_url = state.get('r')
         registry = registry_get(dbname)
         with registry.cursor() as cr:
-            try:
+            #try:
```
Comment (MickeHolm) [inline auth_saml_ol/controllers/main.py]: Remove commented out code

File: auth_saml_ol/controllers/main.py [hunk 2]
```diff
@@ -168,14 +167,17 @@ def signin(self, **kw):
...
+            if True:
                 env = api.Environment(cr, SUPERUSER_ID, context)
                 provider = env['auth.saml.provider'].browse(provider_id)
                 credentials = provider.authenticate(request, kw)
                 request.session['saml_credentials'] = credentials
                 _logger.debug(credentials)
                 res = login_and_redirect(dbname, credentials['login'], 'foobar', redirect_url=redirect_url)
                 return res
+            try:
```
Comment (MickeHolm) [inline auth_saml_ol/controllers/main.py]: Remove do nothing try/except.

File: auth_saml_ol/models/auth_saml.py [hunk 3]
```diff
@@ -177,5 +178,9 @@ def get_saml_user(self, server):
             uid = server.get_nameid()
         else:
             uid = server.get_attribute(self.matching_attribute)
-        return self.env['res.users'].search([('saml_provider_id', '=', self.id), ('saml_uid', '=', uid)])
+        # Why? I didn't need this before.
+        if type(uid) == list:
+            uid = uid[0]
+        _logger.warn(uid)
```
Comment (MickeHolm) [inline auth_saml_ol/models/auth_saml.py]: Remove warning as it look like a debug message as it stands now

File: auth_saml_ol/models/res_users.py [hunk 4]
```diff
@@ -46,7 +46,7 @@ def check_no_password_with_saml(self):
                          'SAML UID must be unique per provider')]
 
     @api.multi
-    def get_saml_data(self, server):
+    def get_saml_data(self, provider, server):
```
Comment (MickeHolm) [inline auth_saml_ol/models/res_users.py:49]: Why add provider if it's not used?

File: auth_saml_ol_groups/models/auth_saml_operator.py [hunk 5]
```diff
@@ -14,19 +14,18 @@ def operators(self):
...
-        matching_value = ''
-        for k in attrs:
-            if isinstance(k, tuple) and mapping.saml_attribute in k:
-                for matching_value in attrs[k]:
-                    if mapping.value in matching_value:
-                        return True
+        values = attrs.get(mapping.saml_attribute)
+        if values:
+            for value in values:
+                #  Does this level even exist in the reply?
```
Comment (MickeHolm) [inline auth_saml_ol_groups/models/auth_saml_operator.py]: Does it exist? 
a few rows down, in "equals", there seems to be no such question?

**Review by MickeHolm — CHANGES_REQUESTED** [review]
I don't like all the "magic numbers" in the code, but I guess that's Odoo standard so...

File: auth_saml_ol/models/res_users.py (hunk 4 above)
Comment (robch2342) [reply auth_saml_ol/models/res_users.py:49]: It's used in auth_saml_ol_groups.

**Pull request merged**